pkg/hubble/relay/observer/server.go

```diff
@@ -81,7 +82,11 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 	ctx, cancel := context.WithCancel(stream.Context())
 	defer cancel()
 
-	peers := s.peers.List()
+	peers, err := filterNodes(req, s.peers.List())
```

Comment:
I am a little confused about the reason to filter the list of peers to which the request gets forwarded to based on the node name filters in Hubble Relay. Could you elaborate on this please?

Reply:
Ah, this looks like a misunderstanding on my part, thanks for catching this.

The goal was to avoid opening connections to peers that will never generate traffic that matches the filter. I was making the assumption that peers and nodes are the same thing, but of course that is not the case: peers can forward flows from other peers and other nodes.

Refactor coming up...